Mwozniak

marcinwozniak0 opened the pull request:
Remove User class and getting 100% coverage of LocalUser

marcinwozniak0 committed:
Remove User class usage from ChatRequest

marcinwozniak0 committed:
Remove inheritacne dependency from User in LocalUser

marcinwozniak0 committed:
Small stylistic refactor in LocalUser class

marcinwozniak0 committed:
Remove and replace User unit tests with LocalUser unit tests

marcinwozniak0 committed:
Remove User class

marcinwozniak0 committed:
Rename test file which testing functionality of LocalUser

marcinwozniak0 committed:
Merge branch 'mwozniak' of https://github.com/NMarcin/MessengerOnFileForLinux into mwozniak

marcinwozniak0 committed:
Minnor changes in LocalUserTest

marcinwozniak0 committed:
Merge branch 'master' of https://github.com/NMarcin/MessengerOnFileForLinux into mwozniak

mnurzyns commented on MessengerOnFileForLinux/ChatStarter/src/ChatRequest.cpp:
```diff
@@ -90,17 +88,17 @@ std::unique_ptr<std::string> ChatRequest::getUserStatus(const std::string& usern
+    std::unique_ptr<std::string> userStatusToCompare = getUserStatus(username);
 
-    UserInactivityDetector userInactivityDetector(user.getUsername()); //TODO mawoznia temportary solution
+    UserInactivityDetector userInactivityDetector(username); //TODO mawoznia temportary solution
```
Z czystej ciekawości co oznacza "temportary solution"?

marcinwozniak0 replied:
Masz tam opisane niżej, UserInactivity bedzie memberem klasy. Nie rozkopywałem tego, bo i tak cala klasa jest do refaktoru i jest na to przygotowany task

mnurzyns commented on MessengerOnFileForLinux/UserService/src/LocalUser.hpp:
```diff
@@ -1,18 +1,24 @@
 private:
     LocalUser();
     ~LocalUser();
     LocalUser(const LocalUser &) = delete;
```
do publica i dodać &&?

marcinwozniak0 replied:
nie, to singleton. Sprawdzałem to wczoraj i tak to musi być

mnurzyns replied:
no to w takim razie chyba wywalić singletona?

marcinwozniak0 replied:
Na razie bym tego nie ruszał, a zrobił na to osobnego taska

mnurzyns replied:
Wywalenie Singletona to banał. Nigdzie tutaj nie tworzysz konfiguracji, tylko zawsze jest stała, bo geteenv. Usuwasz po prostu wszystkie wystąpienia i dodajesz wtedy inicjalizacje klasy.

marcinwozniak0 replied:
Ok, zajme sie tym razem z wyrzuceniem passworda

mnurzyns reviewed (commented):
Minor ...
Jest słabym nazwą commitu bo nie mówi o czym to.

mnurzyns reviewed (dismissed):
Merguj to, ale weź prosze zrób to z password co gadaliśmy

marcinwozniak0 committed:
Add SignOut TestFixture class

marcinwozniak0 committed:
Create new unit test for SignOut

marcinwozniak0 committed:
Add a check that given row to remove exist in file and given row does not exist after remove operation

After this fixes romveRow works corretly. In the past this function always returing true and SignOut did not working correctly

mnurzyns commented on MessengerOnFileForLinux/FileHandling/src/FileHandling.cpp:
```diff
@@ -282,15 +282,26 @@ bool FileInterface::Managment::removeFile(const std::string& pathToFile)
+
+    if (nullptr == Accesor::getRow(pathToFile, pattern))
+    {
+        const std::string logInfo = "FileInterface::Modification::removeRow() -> Attempt to remove unexist row!";
```
static? Czy w sumie zbędne? Ewentualnie jakoś to wybić??

marcinwozniak0 replied:
?

marcinwozniak0 commented on MessengerOnFileForLinux/UserService/test/SignOutTestCore.hpp:
```diff
@@ -0,0 +1,16 @@
+    void TearDown() {}
+    ~SignOutTestFixture() {}
+
+    SignOut signOut;
```
Zastanawiam sie czy nie nazywać takich memberów sut albo cos podobnego, żeby było wiadomo co jest testowane. Tu to róznicy nie robi, ale np. w SignIn już mamy troche więcej memberów w fixturce

mnurzyns replied:
nie rozumiem

mnurzyns commented on MessengerOnFileForLinux/UserService/test/SignOutTestCore.hpp:
```diff
@@ -0,0 +1,25 @@
+#pragma once
```
Po co cała ta klasa skoro nic z tym nie robisz? Nie lepiej w takim razie zrobić z tego po prostu zwykły test file?

marcinwozniak0 replied:
Nie chciałem robić w każdym teście inicjalizcji SignOut no i jeszcze wrzuciłem tu funkcje signInUser(). Uważam że przez to testy są dużo czytelniejsze i od razu widać co robią.

mnurzyns replied:
ok, w takim razie jest drobna uwaga poniżej

mnurzyns commented on MessengerOnFileForLinux/UserService/test/SignOutTestCore.hpp:
```diff
@@ -0,0 +1,25 @@
+    void signInUser()
+    {
+        const std::string USER = getenv("USER");
+        const std::string rowInLoggedFile = "[" + USER + "][0]";
```
Mamy do tego klasę  StringSum

marcinwozniak0 replied:
Remove User class and getting 100% coverage of LocalUser
